src/units.cpp

```diff
@@ -508,55 +497,78 @@ void createUnitMap(const UnitsPtr &units, std::map<std::string, double> &unitMap
                 if (model != nullptr) {
                     auto refUnits = model->units(ref);
                     if ((refUnits == nullptr) || refUnits->isImport()) {
-                        unitMap.clear();
+                        unitsMap.clear();
                         break;
                     }
-                    createUnitMap(refUnits, unitMap, exp); // recursively add units to the map
+                    updateUnitsMap(refUnits, unitsMap, exp);
                 }
             }
         }
     }
 }
 
-bool Units::isEquivalentTo(const UnitsPtr &units1, const UnitsPtr &units2)
+UnitsMap createUnitsMap(const UnitsPtr &units)
+{
+    UnitsMap unitsMap;
+    updateUnitsMap(units, unitsMap);
+    return unitsMap;
+}
+
+bool Units::equivalent(const UnitsPtr &units1, const UnitsPtr &units2)
 {
-    // initial checks
+    // Initial checks.
     if ((units1 == nullptr) || (units2 == nullptr)) {
         return false;
     }
     if ((units1->isImport()) || (units2->isImport())) {
         return false;
     }
 
-    std::map<std::string, double> units1Map;
-    createUnitMap(units1, units1Map);
-    std::map<std::string, double> units2Map;
-    createUnitMap(units2, units2Map); // creating the maps to compare units over
+    UnitsMap units1Map = createUnitsMap(units1);
+    UnitsMap units2Map = createUnitsMap(units2);
 
     if (units1Map.size() == units2Map.size()) {
-        auto it = units1Map.begin();
-
-        while (it != units1Map.end()) {
-            std::string unit = it->first;
+        for (auto &units : units1Map) {
```

Comment:
I'm surprised that there isn't a warning that this should be `const auto &units`.